distributed/shuffle/_arrow.py

```diff
@@ -34,8 +34,8 @@ def check_minimal_arrow_version() -> None:
     Raises a ModuleNotFoundError if pyarrow is not installed or an
     ImportError if the installed version is not recent enough.
     """
-    # First version that supports concatenating extension arrays (apache/arrow#14463)
-    minversion = "12.0.0"
+    # First version to implement type promotion for pa.concat_tables (apache/arrow#36846)
+    minversion = "13.0.0"
```

Comment:
This version came out on Nov 1, 2023 and users are unlikely to have it installed in their environments, so this PR will create some user pain.